tests/pytests/functional/transport/rabbitmq/test_async_pub_channel.py

```diff
@@ -0,0 +1,229 @@
+import ctypes
+import logging
+import multiprocessing
+import signal
+import time
+
+import pytest
+import salt.config
+import salt.exceptions
+import salt.ext.tornado.gen
+import salt.ext.tornado.ioloop
+import salt.log.setup
+import salt.transport.client
+import salt.transport.server
+import salt.utils.platform
+import salt.utils.process
+import salt.utils.stringutils
+from saltfactories.utils.processes import terminate_process
+
+from salt.ext import tornado
+
+log = logging.getLogger(__name__)
+
+
+class Collector(salt.utils.process.SignalHandlingProcess):
+    """
+    A process for collecting published events
+    """
+    def __init__(
+        self, minion_config, pub_uri, aes_key, timeout=30
+    ):
+        super().__init__()
+        self.minion_config = minion_config
+        self.pub_uri = pub_uri
+        self.aes_key = aes_key
+        self.timeout = timeout
+        self.hard_timeout = time.time() + timeout + 30
```

Comment:
**question** why add 30 to the default timeout of 30? :thinking:

Reply:
These tests were adapted from zeromq counterparts. Looking through the code, it looks like "timeout" is used to  break out of consumption loop when no single message was received in "timeout" seconds, while "hard_timeout" is the maximum allowed time to consume all messages. 

I will rename the variables to give them appropriate meaning.